geode-core/src/main/java/org/apache/geode/internal/cache/LocalRegion.java

```diff
@@ -5111,7 +5111,7 @@ public void txApplyPutPart2(RegionEntry regionEntry, Object key, long lastModifi
 
   public boolean basicBridgeCreate(final Object key, final byte[] value, boolean isObject,
       Object callbackArg, final ClientProxyMembershipID client, boolean fromClient,
-      EntryEventImpl clientEvent, boolean throwEntryExists)
+      EntryEventImpl clientEvent, boolean throwEntryExists, boolean generateCallbacks)
       throws TimeoutException, EntryExistsException, CacheWriterException {
```

Comment:
LocalRegion.basicBridgeCreate is only ever called with generateCallbacks=true

The GatewayReceiverCommand create case uses true

The GatewayReceiverCommand update case conditionally LocalRegion.basicBridgePut with false.

I see GatewaySenderEventImpl.initializeAction only sets UPDATE_ACTION or UPDATE_ACTION_NO_GENERATE_CALLBACKS:
```
} else if (operation == EnumListenerEvent.AFTER_UPDATE_WITH_GENERATE_CALLBACKS) {
  if (event.isGenerateCallbacks()) {
    this.action = UPDATE_ACTION;
  } else {
    this.action = UPDATE_ACTION_NO_GENERATE_CALLBACKS;
  }
}
```
That causes only basicBridgePut to be invoked in the GatewayReceiverCommand. Is that intentional? If so, it doesn't look like you need the new parameter on basicBridgeCreate.

Reply:
You are right.